lib/FreeRTOS-Plus-POSIX/source/FreeRTOS_POSIX_clock.c

```diff
@@ -63,11 +63,8 @@ int clock_getcpuclockid( pid_t pid,
     ( void ) pid;
     ( void ) clock_id;
 
-    /* This function is currently unsupported. It will always return -1 and
-     * set errno to EPERM. */
-    errno = EPERM;
-
-    return -1;
+    /* This function is currently unsupported. It will always return EPERM. */
+    return EPERM;
```

Comment:
I didn't know clock_getcpuclockid actually returns errno, but should we still set it for perror to work ?

Reply:
Discussed offline. Leaving it as-is for now, as we are following POSIX original definition of [clock_getcpuclockid()](http://pubs.opengroup.org/onlinepubs/9699919799/functions/clock_getcpuclockid.html). Error number is returned instead of passed via errno variable.  

This brings up a good question of what our POSIX API tests actually check. A good API test case should cover reasonable input combinations, output patterns, and side effects. Clearly we don't do all above all the time...